crates/proof-of-sql-parser/src/posql_time/timestamp.rs

```diff
@@ -6,16 +6,31 @@ use serde::{Deserialize, Serialize};
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
 pub struct PoSQLTimestamp {
     /// The datetime representation in UTC.
-    pub timestamp: DateTime<Utc>,
+    timestamp: DateTime<Utc>,
 
     /// The precision of the datetime value, e.g., seconds, milliseconds.
-    pub timeunit: PoSQLTimeUnit,
+    timeunit: PoSQLTimeUnit,
 
     /// The timezone of the datetime, either UTC or a fixed offset from UTC.
-    pub timezone: PoSQLTimeZone,
+    timezone: PoSQLTimeZone,
```

Comment:
Why did you do that? Getters and setters are usually boilerplate.

Reply:
so that the fields cannot be changed after they are set. i.e. a timestamp parsed as seconds should not be able to be changed to milliseconds without the appropirate scaling